src/services/logsFrame.ts

```diff
@@ -180,3 +190,41 @@ export function getBodyName(logsFrame?: LogsFrame | null): string {
 export function getIdName(logsFrame?: LogsFrame): string {
   return logsFrame?.idField?.name ?? DATAPLANE_ID_NAME;
 }
+
+export function getSeriesVisibleRange(series: DataFrame[]) {
+  let start = 0;
+  let end = 0;
+
+  const timeField = series[0]?.fields.find((field) => field.type === FieldType.time);
+  if (timeField) {
+    const values = timeField.values.sort();
+    const oldestFirst = values[0] < values[values.length - 1];
+    start = oldestFirst ? values[0] : values[values.length - 1];
+    end = oldestFirst ? values[values.length - 1] : values[0];
+  }
+  return { start, end };
+}
+
+export const VISIBLE_RANGE_NAME = 'Visible range';
+export function getVisibleRangeFrame(start: number, end: number) {
+  const frame = arrayToDataFrame([
+    {
+      time: start,
+      timeEnd: start,
+      text: 'Time of the oldest log',
+      color: 'rgb(56, 113, 220, 0.7)',
```

Comment:
nit: can we pull the theme and grab some more theme appropriate colors from there?
<img width="1713" alt="image" src="https://github.com/user-attachments/assets/09d7f81b-2383-4458-b803-9ed618dca335">

Blends in a bit on light mode

Reply:
![2024-12-02 15 31 14](https://github.com/user-attachments/assets/2c4030d6-474f-434a-98bc-6caee028b259)

Looked at the theme but couldn't find a suitable color because I need an rgba with transparency. How do we feel about ☝️ ?

Reply:
Yeah looks much better but I can't see the annotation region on this branch yet?

Reply:
Pushed.